hystrix-core/src/main/java/com/netflix/hystrix/HystrixCommand.java

```diff
@@ -370,7 +399,42 @@ public R execute() {
          * is going to sit waiting on it.
          */
         final Observable<R> o = toObservable();
-        final Future<R> f = o.toBlocking().toFuture();
+        final Future<R> f = new Future<R>() {
+        	final Future<R> base = o.toBlocking().toFuture();
+
+        	@Override
+			public boolean cancel(boolean mayInterruptIfRunning) {
+        		if (base.isCancelled()) {
+        			return false;
+        		}
+
+        		mayInterruptOnCancel.set(mayInterruptIfRunning);
```

Comment:
I think the only valid transition here is false -> true.  If a user does something really weird and gets 2 futures out (f1 and f2), then calls f1.cancel(true); f2.cancel(false), it's unclear about what value would be used by the time `mayInterruptOnCancel` is checked.  I would want the value set to true in that case.

So could you do something like:

```
if (mayInterruptIfRunning) {
    mayInterruptOnCancel.set(true);
}
```